core/src/main/scala/org/apache/spark/deploy/ApplicationDescription.scala

```diff
@@ -20,13 +20,14 @@ package org.apache.spark.deploy
 private[spark] class ApplicationDescription(
     val name: String,
     val maxCores: Option[Int],
-    val memoryPerSlave: Int,
+    val memoryPerExecutorMB: Int,
     val command: Command,
     var appUiUrl: String,
     val eventLogDir: Option[String] = None)
   extends Serializable {
 
   val user = System.getProperty("user.name", "<unknown>")
-
+  // only valid when spark.executor.multiPerWorker is set to true
+  var maxCorePerExecutor = maxCores
```

Comment:
this setting doesn't make sense. `maxCores` is the number of cores shared across all nodes in the cluster, not the number of cores on each executor.